.agents/skills/wpilib-sim/SKILL.md

```
@@ -0,0 +1,218 @@
+- Select sim IO with the repo's established mode behavior when one exists. Team 8044 Rebuilt-2026 defaults desktop runs to `SIM` in `Constants.currentMode`; agents can override with `-DrobotMode=SIM` or `-DrobotMode=REPLAY`.
+- Inspect behavior through live NetworkTables for "what is happening now" and through `*.wpilog` files for repeatable before/after comparisons. Use `$wpilog-decode` for structured WPILOG decoding.
+- Treat AdvantageScope automation as an optional visualization layer. Objective pass/fail checks should still come from WPILOG/NT data unless the user explicitly asks for visual-only output.
+- Do not add per-feature Gradle tasks, scenario classes, or committed harness files unless the user explicitly asks for a persistent test harness.
+- Do not patch robot source with temporary DriverStation/joystick hooks. Rebuilt-2026 has a permanent sim-only `SimAgentBridge`; use it for automated sim control.
+- For automated checks, prefer headless sim: run the normal robot program while disabling GUI/real-driver-station HAL extensions, then inspect NT or WPILOG output.
+
+## Standard Workflow
+
+1. Ground in the repo with `rg`, `git status --short`, and targeted file reads.
+2. Discover the WPILib JDK and any required sim-mode override:
+   ```sh
```

markdownlint-cli2 (0.23.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)